components/authz-service/engine/opa/opa_internal_test.go

```diff
@@ -549,3 +429,864 @@ func randomPolicies(i int) []map[string]interface{} {
+	member := "user:local:test"
+
+	// generate some simple, slightly differentiated policies
+	policies := make(map[string]interface{}, policyCount)
+	for i := 0; i < policyCount; i++ {
+		pid := fmt.Sprintf("pol-%v", i)
+
+		statement := map[string]interface{}{
+			"resources": []string{"*"},
+			"role":      []string{fmt.Sprintf("role-%v", i)},
+			"effect":    "allow",
+			"projects":  []string{fmt.Sprintf("proj-%v", i)},
+		}
+		statements := make(map[string]interface{}, 1)
+		statements["s-1"] = statement
+
+		policies[pid] = map[string]interface{}{
+			"members":    []string{},
+			"statements": statements,
+		}
+	}
+
+	roles := make(map[string]interface{}, roleCount)
+	for j := 0; j < roleCount; j++ {
+		id := fmt.Sprintf("role-%v", j)
+		roles[id] = map[string]interface{}{
+			"actions": []string{fmt.Sprintf("some:automate:action%v", j)},
+		}
+	}
+
+	s.v2p1Store = inmem.NewFromObject(map[string]interface{}{
+		"policies": policies,
+		"roles":    roles,
+	})
+
+	b.Run("store with 0 policies that include the subject as a member", func(b *testing.B) {
+		var resp []string
+		var err error
+		for n := 0; n < b.N; n++ {
+			resp, err = s.V2ProjectsAuthorized(ctx, []string{"user:local:test"}, "iam:projects:delete", "iam:projects", allProjects)
+			if err != nil {
+				b.Error(err)
+			}
+		}
+		projectsResponse = resp
+	})
+
+	for k := 0; k < policyCount; k++ {
+		// add member to each policy as we iterate
+		pol := policies[fmt.Sprintf("pol-%v", k)].(map[string]interface{})
+		pol["members"] = []string{member}
+
+		// refresh store to reflect policies with the subject as a member
+		s.v2p1Store = inmem.NewFromObject(map[string]interface{}{
+			"policies": policies,
+			"roles":    roles,
+		})
+
+		b.Run(fmt.Sprintf("store with %d out of 10 policies that include the subject as a member", k+1), func(b *testing.B) {
+			var resp []string
+			var err error
+			for n := 0; n < b.N; n++ {
+				resp, err = s.V2ProjectsAuthorized(ctx, []string{"user:local:test"}, "iam:projects:delete", "iam:projects", allProjects)
+				if err != nil {
+					b.Error(err)
+				}
+			}
+			projectsResponse = resp
+		})
+	}
+}
+
+// shared (v1/v2) helpers
+func randomTeams(c int) []string {
+	ret := make([]string, c)
+	for i := 0; i < c; i++ {
+		ret[i] = fmt.Sprintf("team:local:team%d", i)
+	}
+	return ret
+}
+
+// v2 helpers
+
+func genericV2Input(subjects []string, resource string, action string, projects []string) (ast.Value, error) {
+	subs := make([]interface{}, len(subjects))
+	for i, sub := range subjects {
+		subs[i] = sub
+	}
+
+	projs := make([]interface{}, len(projects))
+	for i, proj := range projects {
+		projs[i] = proj
+	}
+	input := map[string]interface{}{
+		"subjects": subs,
+		"resource": resource,
+		"action":   action,
+		"projects": projs,
+	}
+	return ast.InterfaceToValue(input)
+}
+
+func v2BaselineAndRandomPoliciesAndRoles(customPolicyCount int, customRoleCount int) (policies map[string]interface{}, roles map[string]interface{}) {
```

Comment:
this helper returns a policy map and role map. 

the returned policy map contains "baseline" policies, AKA everything in the `real_world` json + a number of randomly generated policies.

the returned role map contains chef-managed roles (as defined in the `real_world` json) + a number of randomly generated roles.